trackersite/tracker/tests.py

```diff
@@ -613,351 +757,57 @@ class SummaryTest(TestCase):
-
-class TransactionTest(TestCase):
-    
-    def setUp(self):
-        self.user = User.objects.create(username='user')
-        self.tr = Transaction.objects.create(date=datetime.date(2011, 12, 24), amount=500, other=self.user, description='some desc')
-        
-        # more transactions
-        tr2 = Transaction.objects.create(date=datetime.date(2011, 12, 25), amount=-200, other=self.user, description='other')
-        self.user2 = User.objects.create(username='user2')
-        Transaction.objects.create(date=datetime.date(2011, 12, 26), amount=300, other=self.user2, description='user2')
-    
-    def test_transaction_string(self):        
-        self.assertEqual(u"%s, 500 %s: some desc" % (self.tr.date, settings.TRACKER_CURRENCY), unicode(self.tr))
-    
-    def test_transaction_list(self):
-        c = Client()
-        response = c.get(reverse('transaction_list'))
-        self.assertEqual(200, response.status_code)
-        self.assertEqual(600, response.context['total'])
-    
-    def test_transactions_csv(self):
-        c = Client()
-        response = c.get(reverse('transactions_csv'))
-        self.assertEqual(200, response.status_code)
-        self.assertEqual('text/csv', response['Content-Type'])
-        self.assertEqual(
-            "DATE;OTHER PARTY;AMOUNT CZK;DESCRIPTION;TICKETS;GRANTS;ACCOUNTING INFO\r\n"
-            + "2011-12-26;user2;300.00;user2;;;\r\n"
-            + "2011-12-25;user;-200.00;other;;;\r\n"
-            + "2011-12-24;user;500.00;some desc;;;\r\n", response.content)
-    
-    def test_user_list(self):
-        topic = Topic.objects.create(name='test_topic', ticket_expenses=True, grant=Grant.objects.create(full_name='g', short_name='g'))
-        ticket = Ticket.objects.create(summary='foo', requested_user=self.user2, topic=topic, rating_percentage=100)
-        ticket.add_acks('content', 'docs', 'archive')
-        ticket.expediture_set.create(description='exp', amount=300)
-        ticket.mediainfo_set.create(description='media', count=5)
-        
-        unt = Ticket.objects.create(summary='anon', topic=topic)
-        unt.mediainfo_set.create(description='media', count=3)
-        
-        c = Client()
-        response = c.get(reverse('user_list'))
-        self.assertEqual(200, response.status_code)
-        
-        expected_totals = {
-            'ticket_count': 2,
-            'media': {'objects':2, 'media':8},
-            'accepted_expeditures': 300,
-            'transactions': 600,
-        }
-        self.assertEqual(expected_totals, response.context['totals'])
-        
-        expected_unassigned = {
-            'ticket_count': 1,
-            'media': {'objects':1, 'media':3},
-            'accepted_expeditures': 0,
-        }
-        self.assertEqual(expected_unassigned, response.context['unassigned'])
-
-class ClusterTest(TestCase):
-    def setUp(self):
-        self.user = User.objects.create(username='user')
```

Comment:
Taky řešení :-)

Rád bych tu ale viděl nějaké testy které potvrdí že aktualizace polí `payment_status` a finančních součtů funguje jak má, včetně například interakcí se schváleními. Účetnictví pobočky je dostatečně zábavné i bez chyb zavlečených trackerem.

Taky tu chybí testy importu a exportu.

Reply:
Já to smazal, protože jsem clustery téměř vykostil tak, že nedělají to, co měly a přišlo mi jednodušší tyhle testy smazat a přijít s novými, co pokryjí další funkce.

Reply:
Jo, s tím souhlasím, akorát v kódu a databázi clustery zůstaly, takže vzniklo něco co není pokryté testy a u čeho ani přesně nechápu jak se má chovat.